modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/util/IntrospectionUtils.java

```diff
@@ -247,23 +247,23 @@ public static MetadataType getFieldMetadataType(Field field, ClassTypeLoader typ
     return typeLoader.load(ResolvableType.forField(field).getType());
   }
 
-  public static Field getField(Class<?> clazz, ParameterModel parameterModel) {
+  public static Optional<Field> getField(Class<?> clazz, ParameterModel parameterModel) {
     return getField(clazz, getMemberName(parameterModel, parameterModel.getName()));
   }
 
-  public static Field getField(Class<?> clazz, ParameterDeclaration parameterDeclaration) {
+  public static Optional<Field> getField(Class<?> clazz, ParameterDeclaration parameterDeclaration) {
     return getField(clazz, MuleExtensionAnnotationParser.getMemberName(parameterDeclaration, parameterDeclaration.getName()));
   }
 
-  public static Field getField(Class<?> clazz, String name) {
+  public static Optional<Field> getField(Class<?> clazz, String name) {
     Collection<Field> candidates = getAllFields(clazz, withName(name));
-    return CollectionUtils.isEmpty(candidates) ? null : candidates.iterator().next();
+    return CollectionUtils.isEmpty(candidates) ? Optional.empty() : Optional.of(candidates.iterator().next());
   }
 
-  public static Field getFieldByAlias(Class<?> clazz, String alias) {
+  public static Optional<Field> getFieldByAlias(Class<?> clazz, String alias) {
     Collection<Field> candidates = getAllFields(clazz, withAnnotation(Alias.class));
-    return candidates.stream().filter(f -> alias.equals(f.getAnnotation(Alias.class).value())).findFirst()
-        .orElseGet(() -> getField(clazz, alias));
+    Optional<Field> field = candidates.stream().filter(f -> alias.equals(f.getAnnotation(Alias.class).value())).findFirst();
+    return field.isPresent() ? field : getField(clazz, alias);
```

Comment:
getField return an Optional<Field>. Inside the orElseGet block I must return a Field element.